src/python/bot/tasks/train_gradientfuzz_task.py

```diff
@@ -13,9 +13,308 @@
 # limitations under the License.
 """GradientFuzz training task."""
 
+from builtins import str
+
+import glob
+import os
+import shutil
+import sys
+
+from bot.fuzzers.ml.gradientfuzz import constants
+from bot.fuzzers.ml.gradientfuzz import run_constants
+from build_management import build_manager
+from fuzzing import corpus_manager
+from google_cloud_utils import storage
 from metrics import logs
+from system import archive
+from system import environment
+from system import new_process
+from system import shell
+
+# Model script directory.
+GRADIENTFUZZ_SCRIPTS_DIR = os.path.join(
+    os.path.dirname(os.path.dirname(__file__)), 'fuzzers', 'ml', 'gradientfuzz')
+
+
+def get_corpus(corpus_directory, fuzzer_name):
+  """Get corpus directory.
+
+  This function will download latest corpus backup file from GCS, unzip
+  the file and put them in corpus directory.
+
+  Args:
+    directory: The directory to place corpus.
+    fuzzer_name: Fuzzer name, e.g. libpng_read_fuzzer, xml_parser_fuzzer, etc.
+
+  Returns:
+    True if the corpus can be acquired and False otherwise.
+  """
+  backup_bucket_name = environment.get_value('BACKUP_BUCKET')
+  corpus_fuzzer_name = environment.get_value('CORPUS_FUZZER_NAME_OVERRIDE')
+
+  # Get GCS backup path.
+  gcs_backup_path = corpus_manager.gcs_url_for_backup_file(
+      backup_bucket_name, corpus_fuzzer_name, fuzzer_name,
+      corpus_manager.LATEST_BACKUP_TIMESTAMP)
+
+  # Get local backup path.
+  local_backup_name = os.path.basename(gcs_backup_path)
+  local_backup_path = os.path.join(corpus_directory, local_backup_name)
+
+  # Download latest backup.
+  if not storage.copy_file_from(gcs_backup_path, local_backup_path):
+    logs.log_error(
+        'Failed to download corpus from GCS bucket {}.'.format(gcs_backup_path))
+    return False
+
+  # Extract corpus from zip file.
+  archive.unpack(local_backup_path, corpus_directory)
+  shell.remove_file(local_backup_path)
+
+  return True
+
+
+def get_script_path(which_script):
+  """
+  Gives executable path of requested script.
+
+  Args:
+    which_script (str): Script from `run_constants.py`.
+
+  Returns:
+    (str): Absolute path to requested script (Python file).
+  """
+  return os.path.join(GRADIENTFUZZ_SCRIPTS_DIR, which_script)
+
+
+def get_corpus_directory(root_dir, fuzzer_name):
+  """Gets corpus directory path based on fuzzer."""
+  return os.path.join(root_dir, run_constants.CORPUS_DIR,
+                      fuzzer_name + run_constants.CORPUS_SUFFIX)
+
+
+def get_gcs_model_directory():
```

Comment:
this can be factored out into a common module (as I suggested above) by making the model subdirectory name an argument here.